mri/tests/test_scripts.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+##########################################################################
+# pySAP - Copyright (C) CEA, 2017 - 2018
+# Distributed under the terms of the CeCILL-B license, as published by
+# the CEA-CNRS-INRIA. Refer to the LICENSE file or to
+# http://www.cecill.info/licences/Licence_CeCILL-B_V1-en.html
+# for details.
+##########################################################################
+
+# Package import
+from mri.operators import FFT, WaveletN
+from mri.operators.utils import convert_mask_to_locations
+from mri.reconstructors import SingleChannelReconstructor
+from mri.scripts.gridsearch import launch_grid
+
+# Third party import
+from modopt.math.metrics import ssim
+from modopt.opt.linear import Identity
+from modopt.opt.proximity import SparseThreshold
+import numpy as np
+from pysap.data import get_sample_data
+import unittest
+
+
+class TestScripts(unittest.TestCase):
+    """ Test the scripts in pysap mri like gridsearch
+    """
+    def setUp(self):
+        """ Setup the test variables.
+        """
+        self.n_jobs = 2
+
+    def test_gridsearch_single_channel(self):
+        """Test Gridsearch script in mri.scripts for
+        single channel reconstruction this is a test of sanity
+        and not if the reconstruction is right.
+        """
+        image = get_sample_data('2d-mri')
+        mask = get_sample_data('cartesian-mri-mask')
+        kspace_loc = convert_mask_to_locations(mask.data)
+        fourier_op = FFT(samples=kspace_loc, shape=image.shape)
```

Comment:
if you undersample, there is no guarantee that 0 will be the best regularisation weight.

Reply:
Oh yeah, fixed this and also added a check to ensure that reconstructed image is same as IFFT

Reply:
can you add a little comment specifying that, saying that we don't undersample willignly so that the best `mu` is indeed 0.

Reply:
Oh yeah. on it